### PR TITLE
AoC 2019 Day 1 Part 2

### PR DESCRIPTION
**Pull request by P-on-D**
Implements the updated mass-to-fuel calculation using a recursive function with unit tests based on the example input. Uses the new calculation against the input. Right first time.

### CODE REVIEW

File: 2019/d01p1FuelCounterUpper.d [hunk 1]
```diff
@@ -1,14 +1,23 @@
 import std.algorithm;
 
+int fuelFor(int mass) {
+  int fuel = mass / 3 - 2;
+  return fuel <= 0 ? 0 : fuel + fuelFor(fuel);
```
Comment (P-on-D-Reviewer) [inline 2019/d01p1FuelCounterUpper.d:5]: Why recursion here?

Comment (P-on-D) [reply 2019/d01p1FuelCounterUpper.d:5]: `fuel + fuelFor(fuel)` seems the most natural way to express this. Besides, the compiler knows what I mean and will produce the best code with tail call optimisation.

Comment (P-on-D-Reviewer) [reply 2019/d01p1FuelCounterUpper.d:5]: Point taken on expressiveness but how do you know the compiler is producing the best code?

Comment (P-on-D) [reply 2019/d01p1FuelCounterUpper.d:5]: The awesome and majestic [Compiler Explorer](https://gcc.godbolt.org) pronounced "god bolt". I've prepared some samples.

Compiled with [LDC 1.18.0 unoptimised](https://gcc.godbolt.org/z/_L2rHM) clearly generates recursive code.

Compiled with [LDC 1.18.0 optimised](https://gcc.godbolt.org/z/WjLoEN) frankly I don't know what the fuck it does, but it looks faster.

**Comment by P-on-D-Reviewer** [comment]
LGTM

**Pull request merged**